### PR TITLE
refactor(forum): migrate create and edit views to folio/livewire

### PR DESCRIPTION
**Pull request by wescopeland**
This PR migrates the topic creation route (_createtopic.php_) to Folio, with its form powered by Livewire.

The new route is `/forums/forum/[forum]/topic/create`.

Additionally, the edit form has been migrated to Livewire to ensure functional parity with the create form. The legacy endpoints _forum-topic-comment/update.php_ and _forum-topic/create.php_ have been deleted.

**Before**
Screenshot 2024-05-21 at 6 47 15 PM

**After**
Screenshot 2024-05-21 at 6 46 57 PM

Some fields have been removed because they are redundant with the breadcrumbs and other visual elements on the screen.

### CODE REVIEW

File: resources/views/livewire/forum/create-topic-form.blade.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                name="form.body"
+                label="Body"
+                rows="22"
+                placeholder="Don't share links to copyrighted ROMs."
+                inline
+                required-silent
+                richText
+            >
+                <x-slot name="formActions">
+                    <button type="submit" class="btn">Submit</button>
```
Comment (Jamiras) [inline resources/views/livewire/forum/create-topic-form.blade.php:48]: On submit:
> App\Community\Livewire\Forms\ForumTopicForm::store(): Return value must be of type Illuminate\Http\RedirectResponse, Livewire\Features\SupportRedirects\Redirector returned

The topic does get created.

File: resources/views/livewire/forum/edit-topic-comment-form.blade.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                name="form.body"
+                label="Body"
+                rows="22"
+                placeholder="Don't share links to copyrighted ROMs."
+                inline
+                required-silent
+                richText
+            >
+                <x-slot name="formActions">
+                    <button type="submit" class="btn">Submit</button>
```
Comment (Jamiras) [inline resources/views/livewire/forum/edit-topic-comment-form.blade.php:49]: On edit:
> App\Community\Livewire\Forms\ForumTopicCommentForm::update(): Return value must be of type Illuminate\Http\RedirectResponse, Livewire\Features\SupportRedirects\Redirector returned

The post does get updated

File: resources/views/livewire/forum/create-topic-form.blade.php (hunk 1 above)
Comment (wescopeland) [reply resources/views/livewire/forum/create-topic-form.blade.php:48]: Resolved in latest

File: resources/views/livewire/forum/edit-topic-comment-form.blade.php (hunk 2 above)
Comment (wescopeland) [reply resources/views/livewire/forum/edit-topic-comment-form.blade.php:49]: Resolved in latest

**Review by Jamiras — APPROVED** [review]

**Pull request merged**